### PR TITLE
chore: add types to Map.ts

### PR DESCRIPTION
**Pull request by ojafero**
Completed.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -27,4 +30,126 @@ export type TileLoadingMonitorOptions = {
...
   onLoad: () => void
   onDestroy?: () => void
- }
\ No newline at end of file
+}
+
+//  export type MapOptions = Partial<InstanceType<typeof Map>>
+export type MapOptions = {
+  L: any
+  map?: any
```
Comment (RubenSmn) [inline src/types.ts:38]: This should be `LeafletMap`

**Review by RubenSmn — COMMENTED** [review]
I've found this for the `utfGrid` https://github.com/consbio/Leaflet.UTFGrid/blob/main/L.UTFGrid.js

File: src/types.ts [hunk 2]
```diff
@@ -27,4 +30,126 @@ export type TileLoadingMonitorOptions = {
   onSlowAlert: () => void
   onLoad: () => void
   onDestroy?: () => void
- }
\ No newline at end of file
+}
+
+//  export type MapOptions = Partial<InstanceType<typeof Map>>
+export type MapOptions = {
```
Comment (RubenSmn) [inline src/types.ts:36]: @dadiorchen we created this `MapOptions` for the options you can pass in to the `Map`. Do you know if there are options we have not defined yet?

Comment (dadiorchen) [reply src/types.ts:36]: This is a hard question, maybe this is also the reason why we need the type system, I think it's fine if we missed some type here, we can correct it in the future.

File: src/Map.ts [hunk 3]
```diff
@@ -24,6 +24,35 @@ import Alert from './Alert'
...
+} from './types'
+
+import {
+  Coords,
+  DoneCallback,
+  Marker,
+  TileLayer,
+  Map as LeafletMap,
+  geoJson,
+  GeoJSON,
```
[the remark this answers is not in the document]
Comment (RubenSmn) [reply src/Map.ts:54]: My bad `npm i --save-dev @types/geojson`

```js
import { GeoJsonObject } from 'geojson';
```

**Review by RubenSmn — APPROVED** [review]
Thank you :rocket:

**Pull request merged**